common/src/main/java/net/opentsdb/utils/RefreshingSSLContext.java

```diff
@@ -0,0 +1,800 @@
+// This file is part of OpenTSDB.
+// Copyright (C) 2019  The OpenTSDB Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package net.opentsdb.utils;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.StringReader;
+import java.math.BigInteger;
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.RSAPrivateKeySpec;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.xml.bind.DatatypeConverter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.common.hash.HashCode;
+import com.google.common.hash.Hasher;
+import com.google.common.hash.Hashing;
+import com.google.common.io.Files;
+
+import io.netty.util.Timeout;
+import io.netty.util.TimerTask;
+import net.opentsdb.common.Const;
+import net.opentsdb.core.TSDB;
+import sun.security.provider.certpath.X509CertPath;
+import sun.security.util.DerInputStream;
+import sun.security.util.DerValue;
+
+/**
+ * First stab at a refreshing SSL context that watches the given params for
+ * updates and reloads the context on changes. Note that if the interval is set
+ * to 0 then refreshing is disabled. Makes it useful for servers that load once.
+ * 
+ * @since 3.0
+ */
+public class RefreshingSSLContext implements TimerTask {
```

Comment:
Does the Cert refresher from Athenz not fit the bill?

Reply:
It does, this is just in case we don't have Athenz and eventually I'll make it work like Athenz. I already wrote this code so may as well keep it.

Reply:
Oh and eventually this will be useful for the Undertow context so we can renew the TLS cert without restarting.

Reply:
Athenz Cert refresher would work for Undertow as well. Too much of code. I would recommend not bother maintaining it.

Reply:
No I mean the Athenz refresher is only good for client certs right? Not server certs?